webapp/src/components/HomePage/Map/ParcelsMap/ParcelsMap.js

```diff
@@ -96,6 +97,20 @@ export default class ParcelsMap extends React.Component {
   }
 
   shouldComponentUpdate(nextProps, nextState) {
+    // check if the land data has changed in order to update the popup
+    if (this.popup && this.tileHovered) {
+      const { x, y } = this.tileHovered
+      const coords = buildCoordinate(x, y)
+      const nextParcel = nextProps.parcels[coords]
+      const currentParcel = this.props.parcels[coords]
+      const nextParcelData = (nextParcel && nextParcel.data) || null
+      const currentParcelData = (currentParcel && currentParcel.data) || null
+      const isDifferent = !isEqual(nextParcelData, currentParcelData)
+      if (isDifferent) {
```

Comment:
What do you think of extracting:

```javascript
+   if (this.popup && this.tileHovered) {
+      const { x, y } = this.tileHovered
 +      const coords = buildCoordinate(x, y)
 +      const nextParcel = nextProps.parcels[coords]
 +      const currentParcel = this.props.parcels[coords]
 +      const nextParcelData = (nextParcel && nextParcel.data) || null
 +      const currentParcelData = (currentParcel && currentParcel.data) || null
 +      const isDifferent = !isEqual(nextParcelData, currentParcelData)
```

to it's own method (maybe removing the need for a comment) ?

so we can say 

```javascript
if (this.popupShouldUpdate(/* nextProps.parcels or maybe nextParcel */)) {   // first name that came to mind, not necessarily the best
    this.renderPopup()
}
```